src/components/numeral-date/numeral-date.component.tsx

```diff
@@ -144,12 +270,14 @@ const NumeralDate = ({
     }
   };
 
-  const handleBlur = (datePart) => {
+  const handleBlur = (datePart: DatePart) => {
     const internalValidationEnabled =
       enableInternalError || enableInternalWarning;
     /* istanbul ignore else */
     if (validations[datePart] && internalValidationEnabled) {
-      const errorMessage = validations[datePart](dateValue[datePart])
+      const newDatePart: string =
+        dateValue[datePart as keyof NumeralDateObject];
```

Comment:
Same as the type assertion comment on `Object.fromEntries`

Reply:
The assertion is to satisfy the compiler, as the NumeralDateObject has three types of property combinations.

The date part will always be a key of the NumeralDateObject here, as there is no possibility to have an input that has no corresponding property in that object (please look at invariant errors above).